apps/managedidentity/managedidentity_test.go

```diff
@@ -129,35 +130,155 @@ func Test_SystemAssigned_Returns_Token_Success(t *testing.T) {
+			mockClient.AppendResponse(mock.WithHTTPStatusCode(http.StatusOK), mock.WithBody(responseBody), mock.WithCallback(func(r *http.Request) {
+				localUrl = r.URL
+			}))
 			client, err := New(testCase.miType, WithHTTPClient(&mockClient))
 
 			if err != nil {
 				t.Fatal(err)
 			}
 			result, err := client.AcquireToken(context.Background(), testCase.resource)
-			if !strings.HasPrefix(url, testCase.endpoint) {
-				t.Fatalf("url request is not on %s got %s", testCase.endpoint, url)
+			if !strings.HasPrefix(localUrl.String(), testCase.endpoint) {
+				t.Fatalf("url request is not on %s got %s", testCase.endpoint, localUrl)
+			}
+			if !strings.Contains(localUrl.String(), testCase.miType.value()) {
+				t.Fatalf("url request does not contain the %s got %s", testCase.endpoint, localUrl)
+			}
+			query := localUrl.Query()
+
+			if query.Get(apiVersionQuerryParameterName) != imdsAPIVersion {
+				t.Fatalf("api-version not on %s got %s", imdsAPIVersion, query.Get(apiVersionQuerryParameterName))
+			}
+			if query.Get(resourceQuerryParameterName) != strings.TrimSuffix(testCase.resource, "/.default") {
+				t.Fatal("suffix /.default was not removed.")
+			}
+			switch i := testCase.miType.(type) {
+			case UserAssignedClientID:
+				if query.Get(miQueryParameterClientId) != i.value() {
+					t.Fatalf("resource client-id is incorrect, wanted %s got %s", i.value(), query.Get(miQueryParameterClientId))
+				}
+			case UserAssignedResourceID:
+				if query.Get(miQueryParameterResourceId) != i.value() {
+					t.Fatalf("resource resource-id is incorrect, wanted %s got %s", i.value(), query.Get(miQueryParameterResourceId))
+				}
+			case UserAssignedObjectID:
+				if query.Get(miQueryParameterObjectId) != i.value() {
+					t.Fatalf("resource objectid is incorrect, wanted %s got %s", i.value(), query.Get(miQueryParameterObjectId))
+				}
 			}
 			if err != nil {
 				t.Fatal(err)
 			}
 			if result.AccessToken != token {
 				t.Fatalf("wanted %q, got %q", token, result.AccessToken)
 			}
+
+		})
+	}
+
+	// Testing createIMDSAuthRequest
+	tests := []struct {
+		name     string
+		id       ID
+		resource string
+		claims   string
+		wantErr  bool
+	}{
+		{
+			name:     "System Assigned",
+			id:       SystemAssigned(),
+			resource: "https://management.azure.com",
+		},
+		{
+			name:     "System Assigned",
+			id:       SystemAssigned(),
+			resource: "https://management.azure.com/.default",
+		},
+		{
+			name:     "Client ID",
+			id:       UserAssignedClientID("test-client-id"),
+			resource: "https://storage.azure.com",
+		},
+		{
+			name:     "Resource ID",
+			id:       UserAssignedResourceID("test-resource-id"),
+			resource: "https://vault.azure.net",
+		},
+		{
+			name:     "Object ID",
+			id:       UserAssignedObjectID("test-object-id"),
+			resource: "https://graph.microsoft.com",
+		},
+		{
+			name:     "With Claims",
+			id:       SystemAssigned(),
+			resource: "https://management.azure.com",
+			claims:   "test-claims",
+		},
```

Comment:
This is the only case in this set that adds coverage. When I suggested merging these tests, I meant what you've now done above: validate HTTP requests as sent by `AcquireToken` instead of as returned by `createIMDSAuthRequest`. The connection between public API and HTTP requests matters, not how the client constructs requests.

Given IMDS doesn't support claims and all the other cases starting at line 193 are redundant, I recommend deleting them.